lnwire/typed_fee.go

```diff
@@ -0,0 +1,63 @@
+package lnwire
+
+import (
+	"io"
+
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+const (
+	FeeRecordType tlv.Type = 55555
+)
+
+// Fee represents a fee schedule.
+type Fee struct {
+	BaseFee int32
+	FeeRate int32
+}
+
+// Record returns a TLV record that can be used to encode/decode the fee
+// type from a given TLV stream.
+func (l *Fee) Record() tlv.Record {
+	return tlv.MakeStaticRecord(
+		FeeRecordType, l, 8, feeEncoder, feeDecoder, // nolint: gomnd,lll
+	)
+}
+
+// feeEncoder is a custom TLV encoder for the fee record.
+func feeEncoder(w io.Writer, val interface{}, buf *[8]byte) error {
+	v, ok := val.(*Fee)
+	if !ok {
+		return tlv.NewTypeForEncodingErr(val, "lnwire.Fee")
+	}
+
+	if err := tlv.EUint32T(w, uint32(v.BaseFee), buf); err != nil {
```

Comment:
It amazes me that this works. I can see in the test that it works, but I wonder what's actually encoded in bytes, and whether other implementations can read that as an integer.

Reply:
That cast is really a no-op. The bytes don't change. The encoding for signed and unsigned exists down to the lowest levels in the cpu. https://en.wikipedia.org/wiki/Signed_number_representations#Two's_complement